src/DataGrid.tsx

```diff
@@ -1127,18 +1141,20 @@ function DataGrid<R, SR, K extends Key>(
           onKeyDown={handleKeyDown}
         />
       )}
-      <HeaderRow
-        columns={viewportColumns}
-        onColumnResize={handleColumnResize}
-        allRowsSelected={allRowsSelected}
-        onAllRowsSelectionChange={selectAllRowsLatest}
-        sortColumns={sortColumns}
-        onSortColumnsChange={onSortColumnsChange}
-        lastFrozenColumnIndex={lastFrozenColumnIndex}
-        selectedCellIdx={isHeaderRowSelected ? selectedPosition.idx : undefined}
-        selectCell={selectHeaderCellLatest}
-        shouldFocusGrid={!selectedCellIsWithinSelectionBounds}
-      />
+      <DataGridDefaultComponentsProvider value={defaultHeaderComponents}>
```

Comment:
Wouldn't it be cleaner to just pass `headerRenderer`/`sortIcon` via props?

Reply:
I think I added this avoid adding an extra `sortIcon` prop to `HeaderRenderer`